tiledb/sm/query/query.cc

```diff
@@ -1063,11 +1168,82 @@ Status Query::set_subarray_unsafe(const NDRange& subarray) {
   return Status::Ok();
 }
 
+Status Query::check_buffers_validity() {
```

Comment:
this checks the correctness of buffers for each attribute, let's avoid the word "validity" here because it overloads the "validity" buffer terminology. rename to `check_buffers_correctness`.